File: app/src/main/java/me/blog/hgl1002/openwnn/KOKR/event/SoftKeyEvent.java
package me.blog.hgl1002.openwnn.KOKR.event;

public class SoftKeyEvent extends Event {

	private final SoftKeyAction action;

	private final int keyCode;

	private final SoftKeyPressType type;

	public SoftKeyEvent(SoftKeyAction action, int keyCode) {
		this(action, keyCode, SoftKeyPressType.SIGNLE);
	}

	public SoftKeyEvent(SoftKeyAction action, int keyCode, SoftKeyPressType type) {
		this.action = action;
		this.keyCode = keyCode;
		this.type = type;
	}

	public int getKeyCode() {
		return keyCode;
	}

	public SoftKeyAction getAction() {
		return action;
	}

	public SoftKeyPressType getType() {
		return type;
	}

	public enum SoftKeyAction {
		PRESS, RELEASE, CANCEL;
	}

	public enum SoftKeyPressType {
		SIGNLE, LONG, FLICK_UP, FLICK_DOWN, FLICK_LEFT, FLICK_RIGHT;
	}

}

File: app/src/main/java/io/github/lee0701/heonot/inputmethod/event/SoftKeyEvent.kt
package io.github.lee0701.heonot.inputmethod.event

class SoftKeyEvent @JvmOverloads constructor(
    val action: SoftKeyAction,
    val keyCode: Int,
    val type: SoftKeyPressType = SoftKeyPressType.SINGLE
) :
    Event() {

    enum class SoftKeyAction {
        PRESS, RELEASE, CANCEL
    }

    enum class SoftKeyPressType {
        SINGLE, LONG, FLICK_UP, FLICK_DOWN, FLICK_LEFT, FLICK_RIGHT, REPEAT
    }
}
